### ISSUE TITLE
file.walk "finished" callback

### ISSUE THREAD

**Issue by ryanflorence** [mikeal/node-utils#12]
Would be nice to have a callback for when the entire walk is finished.

``` javascript
file.walk(path, callback, finishedCallback)
```

**Comment by manobi** [comment]
I also need to know when the walk is done, but given that walk already have many arguments my sugestion is:
file.walk(arg1, arg2, arg3).on('end', function(){
   // executed after walk
});

**Comment by ryanflorence** [comment]
a pretty api would be great, just having the ending callback with any api would also be great :)

**Comment by manobi** [comment]
Actually my trick to know when this is finished, is wait for process.on("exit") , but it just work in some contexts.

**Comment by olalonde** [comment]
Right. I don't see the point of having an async walk if there's no way to know when it's finished.... Has this been fixed yet?

**Comment by ghost** [comment]
I agree. This would be very handy to have.

**Comment by Slyke** [comment]
Still not in?

**Comment by cbuteau** [comment]
What are people doing to workaround?  Or should I just write my own utility?

**Comment by michael-ts** [comment]
How is this module even usable without a way to tell when it's done?

**Comment by cbuteau** [comment]
I think I switched to this
https://www.npmjs.com/package/find
I forget it was a script at work.

**Comment by michael-ts** [comment]
@cbuteau haha what a coincidence, that's what i wound up doing too! :-)
